src/environment.rs

```diff
@@ -138,43 +120,83 @@ pub async fn get_up_to_date_prefix(
 
     let mut lock_file = lock_file::load_lock_file(project).await?;
     let up_to_date = lock_file_satisfies_project(project, &lock_file)?;
+    if matches!(usage, LockFileUsage::Locked) && !up_to_date {
+        miette::bail!("Lockfile not up-to-date with the project");
+    }
+    let update_lock = matches!(usage, LockFileUsage::Update) && !up_to_date;
 
-    match usage {
-        LockFileUsage::Update => {
-            if !up_to_date {
-                lock_file = lock_file::update_lock_file(project, lock_file, None).await?
-            }
+    // First lock and install the conda environment
+    // After which we should have a usable prefix to use for pypi resolution.
+    if update_lock {
+        lock_file = lock_file::update_lock_file_conda(project, lock_file, None).await?;
+    }
```

Comment:
Would it be possible to write this as a larger match statement? Or simply use `usage == LockFileUsage::Locked`.

Reply:
Well, I tried that and it results in more duplication or something that is less readable the problem is that the operations depend on each other and the options kind of disable parts of the code that are dependent but at seperate locations.